src/rgw/rgw_sync.cc

```diff
@@ -840,7 +840,7 @@ class RGWFetchAllMetaCR : public RGWCoroutine {
           yield {
             if (!lease_cr->is_locked()) {
               lost_lock = true;
-              break;
+              return 0;
```

Comment:
will try replacing the yield {} block with a single `yield;` statement - then the `break`s will apply to the for loop as intended